File: Python/Compare Version Numbers .py
class Solution:
    # @param version1, a string
    # @param version2, a string
    # @return an integer
    def compareVersion(self, version1, version2):
        list1 = [int(i) for i in (version1.split("."))]
        list2 = [int(i) for i in (version2.split("."))]
        
        length = len(list1) if len(list1) <= len(list2) else len(list2)
        i = 0
        while i< length:
            if list1[i] > list2[i]:
                return 1
            elif list1[i] < list2[i]:
                return -1

            i = i + 1
            
        if len(list1)>length:
            if sum(list1[length:]) != 0:
                return 1
            else:
                return 0
        elif len(list2) > length:
            if sum(list2[length:]) != 0:
                return -1
            else:
                return 0
        else:
            return 0
s = Solution()
print s.compareVersion("1", "1.0")

File: Java/src/Compare Version Numbers.java
public class Solution
{
  public int compareVersion(String version1, String version2)
  {
    String[] list1 = version1.split("\\.");
    String[] list2 = version2.split("\\.");

    int length;
    if (list1.length >= list2.length)
    {
      length = list2.length;
    }
    else
    {
      length = list1.length;
    }

    for (int i = 0; i < length; i++)
    {
      if (Integer.parseInt(list1[i]) > Integer.parseInt(list2[i]))
      {
        return 1;
      }
      else if (Integer.parseInt(list1[i]) < Integer.parseInt(list2[i]))
      {
        return -1;
      }
    }
    if (list1.length > length)
    {
      if (getSum(list1, length) != 0)
      {
        return 1;
      }
      else
      {
        return 0;
      }

    }
    else if (list2.length > length)
    {
      if (getSum(list2, length) != 0)
      {
        return -1;
      }
      else
      {
        return 0;
      }

    }
    else
    {
      return 0;
    }
  }

  public int getSum(String[] list, int length)
  {
    int sum = 0;
    for (int i = length; i < list.length; i++)
    {
      sum = sum + Integer.parseInt(list[i]);
    }
    return sum;
  }
}
